RecoLocalTracker/SiPixelRecHits/python/CustomPixelCPEGeneric.py

```diff
@@ -0,0 +1,29 @@
+'''Customization functions for cmsDriver to change the phase-2 tracker local reconstruction for configuration different from regular planar rectangular pixels'''
+import FWCore.ParameterSet.Config as cms
+
+def esproducers_by_type(process, *types):
+    "Find all ESProducers in the Process that are instances of the given C++ type."
+    return (module for module in process._Process__esproducers.values() if module._TypedParameterizable__type in types)
+
+def customizeFor3DPixels(process):
+    
+    ''' Will remove any usage of template / genError payloads from the reconstruction
+    '''
+    
+    for producer in esproducers_by_type(process, "PixelCPEGenericESProducer"):
+        producer.UseErrorsFromTemplates = False  # no GenErrors
+        producer.LoadTemplatesFromDB = False     # do not load templates
+            
+    return process
+
+def customizeForSquarePixels(process):
+    
+    ''' Do use Template errors for square pixels even in the first tracking step
+        This is needed because hardcoded errors in https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelRecHits/src/PixelCPEGeneric.cc#L113
+        have been optimized for rectangular 25x100 pixels, and in the current generic reco setup we use hardcoded errors for the first tracking pass
+    '''
+    
+    for producer in esproducers_by_type(process, "PixelCPEGenericESProducer"):
+        producer.NoTemplateErrorsWhenNoTrkAngles = cms.bool(False) # use genErrors in the seeding step (when no track angles are available)
```

Comment:
```suggestion
        producer.NoTemplateErrorsWhenNoTrkAngles = False # use genErrors in the seeding step (when no track angles are available)
```
If I'm not mistaken, the parameter is already known (this is just git grep; I did not try to execute)

Reply:
No it's not known. I have tried it and if you do that, when the configuration is being built, when you use 

```
runTheMatrix.py --what upgrade -l 29806.0 -j 0
```

 the framework will complain with:
```
customising the process with customizeForSquarePixels from RecoLocalTracker/SiPixelRecHits/CustomPixelCPEGeneric
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/nweek-02666/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2021-01-31-0000/bin/slc7_amd64_gcc900/cmsDriver.py", line 56, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/nweek-02666/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2021-01-31-0000/bin/slc7_amd64_gcc900/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_11_3_X_2021-02-01-1100/python/Configuration/Applications/ConfigBuilder.py", line 2275, in prepare
    self.pythonCfgCode += self.addCustomise()
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_11_3_X_2021-02-01-1100/python/Configuration/Applications/ConfigBuilder.py", line 885, in addCustomise
    self.process=getattr(package,fcn)(self.process)
  File "/tmp/musich/CMSSW_11_3_X_2021-02-01-1100/python/RecoLocalTracker/SiPixelRecHits/CustomPixelCPEGeneric.py", line 28, in customizeForSquarePixels
    producer.NoTemplateErrorsWhenNoTrkAngles = False # use genErrors in the seeding step (when no track angles are available)
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_11_3_X_2021-02-01-1100/python/FWCore/ParameterSet/Mixins.py", line 276, in __setattr__
    self.__addParameter(name, value)
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_11_3_X_2021-02-01-1100/python/FWCore/ParameterSet/Mixins.py", line 244, in __addParameter
    self.__raiseBadSetAttr(name)
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_11_3_X_2021-02-01-1100/python/FWCore/ParameterSet/Mixins.py", line 299, in __raiseBadSetAttr
    raise TypeError(name+" does not already exist, so it can only be set to a CMS python configuration type")
TypeError: NoTemplateErrorsWhenNoTrkAngles does not already exist, so it can only be set to a CMS python configuration t
```